### ISSUE TITLE
ApplyStep treats environment and namespace as the same

### ISSUE THREAD

**Issue by iocanel** [fabric8io/kubernetes-workflow#31]
So apply step accepts a parameter called 'environment' and uses it as namespace.

In other parts of the ecosystem we treat environments and namespaces differently (env + fabric8.yaml -> namespace) etc.

So I think that if we should either use the same approach in order to find the namespace from the environment or allow explicitly specifying a namespace).

**Comment by iocanel** [comment]
@rawlingsj: I am picking it back, I just need to know if we should try to detect the namespace from fabric8.yaml.

**Comment by rawlingsj** [comment]
Yeah if we can do that it's better. Only I think we'd still need to pass it on from the Jenkinsfile as that's what would decide when to apply to certain environments. 

Issue I had before was that there was no nice way that I could find to parse the fabric8.yml in groovy workflow functions.
